### PR TITLE
Add set HTTP request method

### PR DESCRIPTION
**Pull request by MatkovIvan**

### CODE REVIEW

File: sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/http/DefaultHttpClient.java [hunk 1]
```diff
@@ -158,7 +159,7 @@ private static String doHttpCall(String urlString, String method, Map<String, St
             MobileCenterLog.verbose(LOG_TAG, "HTTP response status=" + status + " payload=" + response);
 
             /* Generate exception on failure. */
-            if (status != 200)
+            if (status != HttpURLConnection.HTTP_OK)
```
Comment (MatkovIvan) [inline sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/http/DefaultHttpClient.java]: Should we use `status >= 200 && status < 300` here?

Comment (jaelim-ms) [reply sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/http/DefaultHttpClient.java]: Good point. We should change that.

File: sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/http/DefaultHttpClient.java [hunk 2]
```diff
@@ -154,8 +154,10 @@ private static String doHttpCall(String urlString, String method, Map<String, St
             MobileCenterLog.verbose(LOG_TAG, "HTTP response status=" + status + " payload=" + response);
 
             /* Generate exception on failure. */
-            if (status != HttpURLConnection.HTTP_OK)
+            boolean successful = status >= 200 && status < 300;
+            if (!successful) {
```
Comment (jaelim-ms) [inline sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/http/DefaultHttpClient.java]: Don't need to define the local variable. 😄 
Please also add unit tests.

Comment (MatkovIvan) [reply sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/http/DefaultHttpClient.java]: Unit tests already here

**Pull request merged**